crates/sncast/src/lib.rs

```diff
@@ -278,15 +279,27 @@ pub async fn wait_for_tx(
     }
     let retries = timeout / u16::from(retry_interval);
     for i in (1..retries).rev() {
-        match provider.get_transaction_receipt(tx_hash).await {
-            Ok(receipt) => {
-                return match receipt.execution_result() {
-                    ExecutionResult::Succeeded => Ok("Transaction accepted"),
-                    ExecutionResult::Reverted { reason } => {
-                        Err(anyhow!("Transaction has been reverted = {reason}"))
+        match provider.get_transaction_status(tx_hash).await {
+            Ok(status) => match status {
+                starknet::core::types::TransactionStatus::Received => {
+                    return Ok("Transaction has been received");
+                }
```

Comment:
I think once transaction is received it still can be rejected or reverted so I guess we shouldn't return here?

Reply:
yup, we should sleep and wait a bit longer, until it changes status.

Reply:
So I will print it only and continue waiting for acceptation.